src/charm.py

```diff
@@ -173,13 +174,31 @@ def _on_config_changed(self, _: ops.HookEvent) -> None:
     def _on_synapse_pebble_ready(self, _: ops.HookEvent) -> None:
         """Handle synapse pebble ready event."""
         self.change_config()
+        # Since Synapse is ready, the charm can create the admin access token
+        container = self.unit.get_container(synapse.SYNAPSE_CONTAINER_NAME)
+        if not container.can_connect():
+            self.unit.status = ops.MaintenanceStatus("Waiting for Synapse pebble")
+            return
+        time.sleep(3)
```

Comment:
Yesterday, I had a completely different idea to remove the dependency between user creation and events. I'll mark this and the following comments as resolved because it has changed a lot by adding a "Token Service". Reviews are welcome! :)